health/health.c

```diff
@@ -577,6 +577,8 @@ void *health_main(void *ptr) {
     time_t now                = now_realtime_sec();
     time_t hibernation_delay  = config_get_number(CONFIG_SECTION_HEALTH, "postpone alarms during hibernation for seconds", 60);
 
+    rrdcalc_labels_unlink();
```

Comment:
If I've read the `rrdcalc_labels_unlink` code correctly then it scans the current labels to decide which alarms to filter out. This can change if the user reloads the labels while the agent is running.

Reply:
In the previous commits I had not arrive in the reload yet, but you are right, we must call it inside health_reload(). I brought this in the last commit.